### PR TITLE
Feat/toggle group

### PR DESCRIPTION
**Pull request by ChrisPaj**
Code for index.html:

<h3>Toggle Button</h3>

    <h4>Toggle Button Default</h4>
    <scale-toggle-button>Click Me!</scale-toggle-button>
    <scale-toggle-button variant="secondary">Click Me!</scale-toggle-button>

    <h4>Icon Before</h4>
    <scale-toggle-button size="regular">
      <scale-icon-action-search></scale-icon-action-search> Label
    </scale-toggle-button>
    <scale-toggle-button variant="secondary" size="regular">
      <scale-icon-action-search></scale-icon-action-search> Label
    </scale-toggle-button>

    <h4>Icon After</h4>
    <scale-toggle-button size="xs">
      Label <scale-icon-action-search></scale-icon-action-search>
    </scale-toggle-button>
    <scale-toggle-button variant="secondary" size="xs">
      Label <scale-icon-action-search></scale-icon-action-search>
    </scale-toggle-button>

    <h4>Icon Only</h4>
    <scale-toggle-button icon-only size="small">
      <scale-icon-action-search></scale-icon-action-search>
    </scale-toggle-button>
    <scale-toggle-button variant="secondary" icon-only size="small">
      <scale-icon-action-search></scale-icon-action-search>
    </scale-toggle-button>

    <h4>Disabled</h4>
    <scale-toggle-button size="xs" disabled>
      Label <scale-icon-action-search></scale-icon-action-search>
    </scale-toggle-button>
    <scale-toggle-button variant="secondary" size="xs" disabled>
      Label <scale-icon-action-search></scale-icon-action-search>
    </scale-toggle-button>

    <h3>Toggle Group</h3>
    <h4>Default (Large)</h4>
    <scale-toggle-group>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button selected>Click Me!</scale-toggle-button>
    </scale-toggle-group>

    <h4>Small Secondary</h4>
    <scale-toggle-group size="small" variant="secondary">
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button selected>Click Me!</scale-toggle-button>
    </scale-toggle-group>

    <h4>XS Block</h4>
    <scale-toggle-group size="xs" box-type="block" >
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button selected>Click Me!</scale-toggle-button>
    </scale-toggle-group>

    <h4>Regular Secondary Block Border</h4>
    <scale-toggle-group size="regular" box-type="block" variant="secondary" border>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button selected>Click Me!</scale-toggle-button>
    </scale-toggle-group>

    <h4>Disabled Border</h4>
    <scale-toggle-group disabled border>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button selected>Click Me!</scale-toggle-button>
    </scale-toggle-group>

    <h4>Regular Non-Multi</h4>
    <scale-toggle-group multi="false">
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button>Click Me!</scale-toggle-button>
      <scale-toggle-button selected>Click Me!</scale-toggle-button>
    </scale-toggle-group>

### CODE REVIEW

File: packages/components/src/components/toggle-button/toggle-button.css [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+:host {
+  --width: auto;
+  --spacing-x: var(--scl-spacing-24);
+  --spacing-x-icon-only: var(--scl-spacing-8);
+  --min-height: var(--scl-spacing-24);
+  --xs-height: var(--scl-spacing-24);
+  --small-height: var(--scl-spacing-32);
```
Comment (nowseemee) [inline packages/components/src/components/toggle-button/toggle-button.css:18]: the pattern we are trying to follow is: `--cssProperty-thing-state`
so like the following:
```
--height-small
--height-regular
...
```

File: packages/components/src/components/toggle-button/toggle-button.css [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+  font-weight: var(--font-weight);
+  font-size: var(--font-size);
+  line-height: var(--line-height);
+  min-height: var(--min-height);
+  width: var(--width);
+  padding-left: var(--spacing-x);
+  padding-right: var(--spacing-x);
+  vertical-align: var(--vertical-align);
+  transition: var(--transition);
+  background: #fff;
```
Comment (nowseemee) [inline packages/components/src/components/toggle-button/toggle-button.css:84]: do you really need a white background?

File: packages/components/src/components/toggle-button/toggle-button.tsx [hunk 3]
```diff
@@ -0,0 +1,158 @@
...
+  shadow: true,
+})
+export class ToggleButton {
+  @Element() hostElement: HTMLElement;
+  /** (optional) The size of the button */
+  @Prop() size?: 'large' | 'regular' | 'small' | 'xs' = 'large';
+  /** (optional) Button variant */
+  @Prop() variant?: 'primary' | 'secondary' = 'primary';
+  /** (optional) background color scheme of a selected button */
+  @Prop() colorScheme?: 'magenta' | 'black' = 'magenta';
```
Comment (nowseemee) [inline packages/components/src/components/toggle-button/toggle-button.tsx:44]: let's avoid using magenta - we could do primary/secondary or light/dark?

File: packages/components/src/components/toggle-button/toggle-button.tsx [hunk 4]
```diff
@@ -0,0 +1,158 @@
...
+  }
+
+  componentWillLoad() {
+    if (this.toggleButtonId == null) {
+      this.toggleButtonId = 'toggle-button-' + i++;
+    }
+  }
+
+  handleIconSize() {
+    const icon = this.hostElement.children[0] && this.hostElement.children[0];
```
Comment (nowseemee) [inline packages/components/src/components/toggle-button/toggle-button.tsx:84]: what if `this.styles` is truthy? then the first child will be <styles>, right?

Comment (nowseemee) [reply packages/components/src/components/toggle-button/toggle-button.tsx:84]: what if `this.styles` is truthy? then the first child will be `<styles>...</styles>`, right?

File: packages/components/src/components/toggle-group/toggle-group.css [hunk 5]
```diff
@@ -0,0 +1,22 @@
+:host {
+  --border-color: #262626;
```
Comment (nowseemee) [inline packages/components/src/components/toggle-group/toggle-group.css:2]: please pick a color from the tokens

File: packages/components/src/components/toggle-group/toggle-group.tsx [hunk 6]
```diff
@@ -0,0 +1,167 @@
...
+  @Prop() multi: boolean = true;
+  /** (optional) aria-label attribute needed for icon-only buttons */
+  @Prop() ariaLabel: string;
+  /** (optional) Injected CSS styles */
+  @Prop() styles?: string;
+  /** Emitted when button is clicked */
+  @Event() scaleClickGroup!: EventEmitter;
+
+  @Listen('scaleClick')
+  scaleCickHandler(ev) {
```
Comment (nowseemee) [inline packages/components/src/components/toggle-group/toggle-group.tsx:59]: nit: there is typo, you miss an `l`

File: packages/components/src/components/toggle-group/toggle-group.tsx [hunk 7]
```diff
@@ -0,0 +1,167 @@
...
+      const button = toggleButton.shadowRoot.querySelector('button');
+      tempState.push({
+        id: button.getAttribute('id'),
+        selected: toggleButton.hasAttribute('selected'),
+      });
+      toggleButton.setAttribute('size', this.size);
+      toggleButton.setAttribute('variant', this.variant);
+      toggleButton.setAttribute('disabled', this.disabled && 'disabled');
+    });
+    // console.log('tempState', tempState)
```
Comment (nowseemee) [inline packages/components/src/components/toggle-group/toggle-group.tsx:99]: let's remove the dead code

File: packages/components/src/components/toggle-button/toggle-button.css (hunk 1 above)
Comment (ChrisPaj) [reply packages/components/src/components/toggle-button/toggle-button.css]: css naming has beed adapted...

File: packages/components/src/components/toggle-button/toggle-button.css (hunk 2 above)
Comment (ChrisPaj) [reply packages/components/src/components/toggle-button/toggle-button.css]: no need for this background. I have deleted it ;-)

File: packages/components/src/components/toggle-button/toggle-button.tsx (hunk 4 above)
Comment (ChrisPaj) [reply packages/components/src/components/toggle-button/toggle-button.tsx:84]: <styles>...</styles> will not be the first child, when being applied. I think the children are the slotted elements (text and/or icon and 'unwished' text nodes due tue formatting, which are eliminated).

File: packages/components/src/components/toggle-group/toggle-group.css (hunk 5 above)
Comment (ChrisPaj) [reply packages/components/src/components/toggle-group/toggle-group.css]: ok, everything with tokens now...

File: packages/components/src/components/toggle-group/toggle-group.tsx (hunk 7 above)
Comment (ChrisPaj) [reply packages/components/src/components/toggle-group/toggle-group.tsx]: dead code removed...

**Comment by acstll** [comment]
I'm gonna leave this one for Daniel for next week 😁

**Pull request merged**